File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    same, not_same = 0, 0
    for c in s:
        if same == not_same:
            answer += 1
            same, not_same = 0, 0
            x = c
            
        if x == c:
            same += 1
        elif x != c:
            not_same += 1
            
    return answer

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.swift
import Foundation

func solution(_ s:String) -> Int {
    var arr = Array(s).map { String($0) }
    var result = 0
    var (same, notSame) = (0, 0)
    var x = ""
    // ca bba
    for c in arr {
        if same == notSame {
            result += 1
            same += 1
            x = c
        } else {
            if c == x {
                same += 1
            } else {
                notSame += 1
            }
        }
    }
    
    return result
}